src/cpp/rtps/persistence/SQLite3PersistenceService.cpp

```diff
@@ -206,7 +210,15 @@ SQLite3PersistenceService::~SQLite3PersistenceService()
     finalize_statement(load_reader_stmt_);
     finalize_statement(update_reader_stmt_);
 
-    sqlite3_close(db_);
+    // Finalize writer seq_num statements
+    finalize_statement(load_writer_last_seq_num_stmt_);
+    finalize_statement(update_writer_last_seq_num_stmt_);
+
+    int res = sqlite3_close(db_);
+    if(res != SQLITE_OK) // (0) SQLITE_OK
+    {
+        logWarning(RTPS_PERSISTENCE, "Database could not be closed. sqlite3_close code: " << res);
```

Comment:
I'd make this an ERROR log, as we suspect it is the cause of the issues we are experiencing, and we already saw it may affect the user.